admin-front/src/components/Users/Users.tsx

```diff
@@ -79,33 +64,46 @@ export const UserEdit = ({ ...props }) => {
     refresh();
   };
 
-  const [roles, setRoles] = React.useState([])
+  const [roles, setRoles] = useState([]);
 
-  React.useEffect(() => {
-    axios.get(process.env.REACT_APP_API_SERVER_BASE_URL + '/role', {
-      headers: {
-        'Authorization': `Bearer ${localStorage.getItem('KanvasAdmin - Bearer')}`
-      }
-    })
-      .then(response => {
-        setRoles(response.data.data.map((role: any) => ({ id: role.id, name: role.role_label.charAt(0).toUpperCase() + role.role_label.slice(1) })))
-      }).catch(error => {
-        notify(`An error occured while fetching the roles`);
-        console.log(error)
+  useEffect(() => {
```

Comment:
I'm not very acquainted with React but, can't this also be replaced with a `const { roles } = UseGetRolesFromAPI();` like you did above?

And same point a bit below.

Reply:
You are correct. NftList is using a different format of "roles" than "NftEdit" or "NftCreate". But should look cleaner now